src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -65,7 +66,13 @@
 @Extension public class LibraryAdder extends ClasspathAdder {
 
     private static final Logger LOGGER = Logger.getLogger(LibraryAdder.class.getName());
+    
+    private static ConcurrentHashMap<String, ReentrantReadWriteLock> cacheRetrieveLock = new ConcurrentHashMap<>();
```

Comment:
This will over time grow unbounded, however I not expecting a lot of changes in the name here [(name, version, trusted, source)](https://github.com/jenkinsci/pipeline-groovy-lib-plugin/blob/2b196c74569a90bfa0977eef6152b540cdd9a93f/src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryRecord.java#L65) so using caffeine or `WeakHashMap` with synchronization is probably overkill (let alone the potential for premature eviction due to the use of a non interned string as a key).